actor/v2action/service_access.go

```diff
@@ -0,0 +1,225 @@
+	}
+
+	servicePlans, warnings, err := actor.CloudControllerClient.GetServicePlans(ccv2.Filter{
+		Type:     constant.ServiceGUIDFilter,
+		Operator: constant.EqualOperator,
+		Values:   []string{services[0].GUID},
+	})
+
+	allWarnings = append(allWarnings, warnings...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	for _, plan := range servicePlans {
+		warnings, err = actor.CloudControllerClient.UpdateServicePlan(plan.GUID, true)
+		allWarnings = append(allWarnings, warnings...)
+		if err != nil {
+			return allWarnings, err
+		}
+	}
+
+	return allWarnings, nil
+}
+
+// EnablePlanForAllOrgs enables access to a specific plan of the given service in all orgs.
+func (actor Actor) EnablePlanForAllOrgs(serviceName, servicePlanName string) (Warnings, error) {
+	var allWarnings Warnings
+
+	services, warnings, err := actor.CloudControllerClient.GetServices(ccv2.Filter{
+		Type:     constant.LabelFilter,
+		Operator: constant.EqualOperator,
+		Values:   []string{serviceName},
+	})
+	allWarnings = append(allWarnings, warnings...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	if len(services) == 0 {
+		return allWarnings, actionerror.ServiceNotFoundError{Name: serviceName}
+	}
+
+	servicePlans, warnings, err := actor.CloudControllerClient.GetServicePlans(ccv2.Filter{
+		Type:     constant.ServiceGUIDFilter,
+		Operator: constant.EqualOperator,
+		Values:   []string{services[0].GUID},
+	})
+	allWarnings = append(allWarnings, warnings...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	// We delete all service plan visibilities for the given Plan since the attribute public should function as a giant on/off
+	// switch for all orgs. Thus we need to clean up any visibilities laying around so that they don't carry over.
+	for _, plan := range servicePlans {
+		if plan.Name == servicePlanName {
+			warnings, err := actor.removeOrgLevelServicePlanVisibilities(plan.GUID)
+			allWarnings = append(allWarnings, warnings...)
+			if err != nil {
+				return allWarnings, err
+			}
+
+			ccv2Warnings, err := actor.CloudControllerClient.UpdateServicePlan(plan.GUID, true)
+			allWarnings = append(allWarnings, ccv2Warnings...)
+			return allWarnings, err
+		}
+	}
+
+	return allWarnings, actionerror.ServicePlanNotFoundError{PlanName: servicePlanName, ServiceName: serviceName}
+}
+
+// EnableServiceForOrg enables access for the given service in a specific org.
+func (actor Actor) EnableServiceForOrg(serviceName, organization string) (Warnings, error) {
+	var allWarnings Warnings
+
+	services, warnings, err := actor.CloudControllerClient.GetServices(ccv2.Filter{
+		Type:     constant.LabelFilter,
+		Operator: constant.EqualOperator,
+		Values:   []string{serviceName},
+	})
+	allWarnings = append(allWarnings, warnings...)
+
+	if err != nil {
+		return allWarnings, err
+	}
+
+	if len(services) == 0 {
+		return nil, actionerror.ServiceNotFoundError{Name: serviceName}
+	}
+
+	orgs, warnings, err := actor.CloudControllerClient.GetOrganizations(ccv2.Filter{
```

Comment:
I think it would be better if you use `actor.GetOrganizationByName(orgName)` instead of  relying on `actor.CloudControllerClient.GetOrganziations(...)`. 

* It would remove some code duplication. Don't have to check `len(orgs)`
* The actor method also throws an exception if multiple organization with the same name are found (MultipleOrganizationsFoundError). This scenario is not possible via cf create-org but there might be other ways of creating this scenario (suppose apps-manager accidentally allows this possibility).